Library/AddOn.cs

```diff
@@ -231,6 +238,41 @@ internal override void WriteXml(XmlTextWriter xmlWriter)
             xmlWriter.WriteEndElement();
         }
 
+        internal void WriteTieredUpdateXml(XmlTextWriter xmlWriter) {
```

Comment:
Does it make sense to have a separate method for this when most of the values are the same? Worried it might result in people applying serializer updates to the wrong method or only one of them.

Reply:
Alternatively, I could alter `WriteXml()` to include the conditional below:
```c#
if (this.TierType == "flat")
  xmlWriter.WriteIfCollectionHasAny("unit_amount_in_cents", UnitAmountInCents, pair => pair.Key,
  pair => pair.Value.AsString());
```
That minor change rids us of the error `currencies cannot be used with tiers` that gets thrown when invoking `WriteXml()` in `Update()` in instances where `tier_type != flat`.